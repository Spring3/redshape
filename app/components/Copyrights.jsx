import React from 'react';
import styled from 'styled-components';

import Link from './Link';

const Copyrights = styled.p`
  color: #FF7079;
`;

export default () => (
  <Copyrights>
    Created by&nbsp;
    <Link
      type="external"
      href="https://spring3.github.io/website/"
    >
      Daniyil Vasylenko
    </Link>
  </Copyrights>
);
