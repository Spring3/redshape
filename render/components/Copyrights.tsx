import React from 'react';
import { css } from '@emotion/react';

import { Link } from './Link';

const Copyrights = () => (
  <p css={css`color: #FF7079`}>
    Created by&nbsp;
    <Link
      external
      href="https://www.dvasylenko.com/redshape"
      testId="copyrights-link"
    >
      danv
    </Link>
  </p>
);

export {
  Copyrights
};
