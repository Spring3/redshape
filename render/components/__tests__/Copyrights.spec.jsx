import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Copyrights from '../Copyrights';

describe('Copyrights Component', () => {
  it('should render the ref to the website', () => {
    const { queryByText, getByTestId } = render(<Copyrights />);
    expect(queryByText('Created by')).toBeDefined();
    const copyrightsLink = getByTestId('copyrights-link');
    expect(copyrightsLink).toBeInTheDOM();
    expect(copyrightsLink).toHaveAttribute('href', 'https://www.dvasylenko.com');
  });
});
