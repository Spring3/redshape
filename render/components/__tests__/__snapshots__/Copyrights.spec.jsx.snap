// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copyrights Component should match the snapshot 1`] = `
.c1 {
  font-size: 14px;
  padding: 2px;
  cursor: pointer;
}

.c1:hover {
  background: undefined;
  color: undefined;
}

.c0 {
  color: #FF7079;
}

<p
  className="c0"
>
  Created byÂ 
  <a
    className="c1"
    href="https://spring3.github.io/website/"
    onClick={[Function]}
    rel="noopener noreferer"
    type="external"
  >
    Daniyil Vasylenko
  </a>
</p>
`;
